### DISCUSSION TITLE
Python Virtual Environment

### DISCUSSION THREAD

**Discussion by tyeeman** [squidfunk/mkdocs-material#4587] [Q&A]
I've been running mkdocs-material for a few weeks from my main python install folder which I read is not the thing to do. So I installed it in a virtual environment and uninstalled it from the main python folder. Now the question is can I run it from another random folder by specifying the mkdocs.exe that sits in that virtual folder? Here is my batch file and it does run just fine - 

```
"C:\PythonVenvProjects\mkdocs-material\env\Scripts\mkdocs.exe" serve
```
So is this the correct way to run it, without activating the virtual environment first?

**Comment by squidfunk** [comment]
I always use the following steps when working with virtual environments, which you use to isolate one's project's dependencies from another one. First, change to the folder of your project. Then:

``` sh
# Create the virtual environment in the folder 'venv'
python -m venv venv

# Activate the virtual environment – your shell should now be prefixed with (venv)
. ./venv/bin/activate
```

Now, all calls to `pip` will install dependencies in the `venv` folder. To exit the virtual environment, use:

```
deactivate
```

**Reply by tyeeman** [reply]
Yes I ran those commands first. Then how to run the serve command from a batch file using the venv folder? That was my command line in the first post just needing confirmation. Once all dependencies are installed in the venv, we don't have to actually activate the venv anymore just to run "mkdocs serve" do we?
